src/Forms/Form/index.js

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { merge, omit, reduce, get } from 'lodash/fp';
+import withStyles from '../../../util/withStyles';
+import styles from './index.scss';
+
+export const FORM_CONTEXT = {
+  form: PropTypes.shape({
+    data: PropTypes.object,
+    country: PropTypes.string,
+    onChange: PropTypes.func,
+    onFieldChange: PropTypes.func,
+    onSubmit: PropTypes.func
+  }).isRequired
+};
+
+export function onChangeForm(prevForm, form) {
+  return merge(prevForm, form);
+}
+
+export function createFormState(
+  fields,
+  initialValues = {
+    values: {},
+    meta: {},
+    metaPaths: {},
+    validations: {},
+    messages: {}
+  }
+) {
+  return reduce(
+    (memo, field) => {
+      const { values, metaPaths, validations, messages } = initialValues;
+      memo.values[field] = get(field, values) || '';
+      memo.errors[field] = {};
+      memo.dirty[field] = false;
+      memo.meta[field] = {};
+      memo.metaPaths[field] = get(field, metaPaths) || '';
+      memo.validations[field] = get(field, validations) || {};
+      memo.messages[field] = get(field, messages) || {};
+      return memo;
+    },
+    {
+      values: {},
+      errors: {},
+      dirty: {},
+      meta: {},
+      metaPaths: {},
+      validations: {},
+      messages: {},
+      valid: false
```

Comment:
Why do we not rename this to `isValid`. Been wondering that for a while. Does it conflict with something else? I think it's more intuitive for a boolean.